### ISSUE TITLE
Printing image alignment issue

### ISSUE THREAD

**Issue by saad3344555** [CesarBalzer/Cordova-Plugin-BTPrinter#61]
I have aligned the image to the right in java file before executing the print command. It prints with correct alignment at the first time. For second time, image starting position is moved somehow and the cropped part appears in the beginning. This continues and position keeps on changing. My question is there a reset command that can may be tell the printer to fix the position and alignment to default again

outputStream.write(setBytes.ESC_ALIGN_LEFT); // left alignment 
outputStream.write(command); //printing image byte array with GS v 0
outputStream.write(setBytes.FEED_PAPER_AND_CUT) //feed or cut paper 
outputStream.flush() //flushing stream

This piece of code gets called every time for consecutive prints but printer acts weird

@CesarBalzer  @andreszs  @ochakov @nicolaszwier

**Comment by andreszs** [comment]
Have you tried sending a printer's [reset command](https://github.com/andreszs/cordova-plugin-demos/blob/e9a25d7f35d6c2e67b5318e671230dd843bbd4bb/com.andreszs.btprinter.demo/www/js/index.js#L296) as shown in the demo app? Most image problems are related to the printer's cache.

**Comment by saad3344555** [comment]
Yes @andreszs but it doesn't work at all. Unless i power off the printer ill get same pattern.

**Comment by andreszs** [comment]
Why don't you share a photo of the sample output ticket to help you think a workaround for this?

**Comment by saad3344555** [comment]
![1655294784780](https://user-images.githubusercontent.com/20259664/173823428-d7d5f410-a10e-4446-a16b-0286a1c29455.JPEG)

**Comment by saad3344555** [comment]
First image is printed correct.

**Comment by andreszs** [comment]
All I can suggest at this time is that you reorder the POS commands sent, just like you would reorder CSS floats in a webpage.

If everything else fails, try to convert the entire header text + image to a **100% width image** and print that image normally without alignment.

**Comment by saad3344555** [comment]
Actually everything here is an image. This is basically a single image. I want to know if there is an esc command for wiping out printer's cache or memory except INIT which is 'ESC @'

**Comment by saad3344555** [comment]
Also the final bytes sent to printer for printing command are also the identical

**Comment by andreszs** [comment]
> Actually everything here is an image. This is basically a single image. I want to know if there is an esc command for wiping out printer's cache or memory except INIT which is 'ESC @'

You can either check the printer's manual or the generic [EPSON ESC/POS](https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=2) commands reference for that.

**Comment by saad3344555** [comment]
Actually i faced issue on SEWOO printer. So I replaced the image bit conversion code to the one used in the android sdk of 
SEWOO and it worked like a charm

**Comment by saad3344555** [comment]
How ever, I find one unusual thing in the existing plugin java code which is basically the GS v 0 args.
The width and height strings are computed wrong i guess or i think it will work for smaller images rather than bigger images for e.g. 1022 x 2024

**Comment by andreszs** [comment]
Sending large images and expecting the printer or the plugin to auto-shrink them is a common mistake, yes. Whenever you do such a thing you are looking for problems. Always resize the images to the smallest size that will result in an acceptable quality.

**Comment by saad3344555** [comment]
Yes you are right but the issue is why it was kind of auto shifting the bytes on repetitive writes to printer stream. I tackled the image size part previously else it wouldn't have printed at first place.
